catalog/imaging/imaging-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,396 @@
+            nitfPostProcessPlugin.processCreate(null)
+        then: "the plugin should throw an exception"
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling bad input during update requests"() {
+        when: "the plugin receives a null input for an update request"
+            nitfPostProcessPlugin.processUpdate(null)
+        then: "the plugin should throw an exception"
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling bad input during delete requests"() {
+        when: "the plugin receives a null input for a delete request"
+            nitfPostProcessPlugin.processDelete(null)
+        then: "the plugin should throw an exception"
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling basic create requests"() {
+        when: "a valid process create request is submitted"
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            2 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value != null})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+            1 * processCreateItem.markMetacardAsModified()
+            createProcessRequest.getProcessItems().size() == 3
+    }
+
+    def "handling basic update requests"() {
+        when: "a valid process update request is submitted"
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            2 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value != null})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+            1 * processUpdateItem.markMetacardAsModified()
+            updateProcessRequest.getProcessItems().size() == 3
+    }
+
+    def "handling processing large files in create requests"() {
+        setup:
+            nitfPostProcessPlugin.setMaxNitfSizeMB(4)
+        when: "a request contains a resource that is too large to process"
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then: "nothing should be added to the metacard"
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling processing large files in update requests"() {
+        setup:
+            nitfPostProcessPlugin.setMaxNitfSizeMB(4)
+        when: "a request contains a resource that is too large to process"
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling create requests with overview disabled"() {
+        setup:
+            nitfPostProcessPlugin.setCreateOverview(false)
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("original")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
```

Comment:
Do we also need to assert that overview wasn't set here? Same comment for the next few methods